test/bond-mediator.test.ts

```diff
@@ -31,7 +31,7 @@ import {ExtendedERC20} from './contracts/cast/extended-erc20'
 chai.use(solidity)
 
 describe('Bond Mediator contract', () => {
-    before(async () => {
+    beforeEach(async () => {
```

Comment:
question: what's the need for this change?

As a pattern, let's keep the scope of `beforeEach` minimal i.e.  by using another `describe` nesting. 

_Although fine in small tests, when we scale (larger setups), I've experienced the test execution time causing much frustration_

Reply:
I guess to prevent interleaving between test logic, ironically I always enjoy watching tests run! 
Right I'll revert back to `before` and use a new `describe`!

Reply:
Haha, I agree, it can be satisfying ...when they get to >5mins, I do feel the impatience seeping in 😅

Reply:
I've switched to having an `unpause()` in `before` describe - I'm not sure if this is the best way to do this? Other solution: unpause after the 'only when not paused' tests in the previous block?

Reply:
If `unpause()` in the `before` works then go with that 👍

Reply:
It's a nice middle ground between isolated tests (they're not building atop of each other's state) and minimal execution time (due to test setup time)